src/Paket.Core/Installation/InstallProcess.fs

```diff
@@ -37,35 +37,38 @@ let updatePackagesConfigFile (model: Map<GroupName*PackageName,SemVerInfo*Instal
-                let di = DirectoryInfo(Path.Combine(root, Constants.PackagesFolderName, groupName))
-                if di.Exists then di else
-
-                match di.GetDirectories() |> Seq.tryFind (fun subDir -> String.endsWithIgnoreCase groupName subDir.FullName) with
-                | Some x -> x
-                | None ->
-                    traceWarnfn "The following directories exists:"
-                    di.GetDirectories() |> Seq.iter (fun d -> traceWarnfn "  %s" d.FullName)
-
-                    failwithf "Group directory for group %s was not found." groupName
-
-        match di.GetDirectories() |> Seq.tryFind (fun subDir -> String.endsWithIgnoreCase lowerName subDir.FullName) with
-        | Some x -> x
-        | None ->
-            traceWarnfn "The following directories exists:"
-            di.GetDirectories() |> Seq.iter (fun d -> traceWarnfn "  %s" d.FullName)
-
-            failwithf "Package directory for package %O was not found." packageName
+    let storageOption = defaultArg settings.StorageConfig PackagesFolderGroupConfig.Default
+    match storageOption.Resolve root groupName packageName version includeVersionInPath with
+    | ResolvedPackagesFolder.ResolvedFolder targetFolder (*when Directory.Exists targetFolder*) ->
+        DirectoryInfo targetFolder
+    | ResolvedPackagesFolder.NoPackagesFolder ->
+        let d = DirectoryInfo(NuGetCache.GetTargetUserFolder packageName version)
+        if not d.Exists then failwithf "Package directory for package %O was not found." packageName
+        d
+        //let lowerName = packageName.ToString() + if includeVersionInPath then "." + version.ToString() else ""
+        //let di =
+        //    if groupName = Constants.MainDependencyGroup then
+        //        DirectoryInfo(Path.Combine(root, Constants.PackagesFolderName))
+        //    else
+        //        let groupName = groupName.CompareString
+        //        let di = DirectoryInfo(Path.Combine(root, Constants.PackagesFolderName, groupName))
+        //        if di.Exists then di else
+        //
+        //        match di.GetDirectories() |> Seq.tryFind (fun subDir -> String.endsWithIgnoreCase groupName subDir.FullName) with
+        //        | Some x -> x
+        //        | None ->
+        //            traceWarnfn "The following directories exists:"
+        //            di.GetDirectories() |> Seq.iter (fun d -> traceWarnfn "  %s" d.FullName)
+        //
+        //            failwithf "Group directory for group %s was not found." groupName
+        //
+        //match di.GetDirectories() |> Seq.tryFind (fun subDir -> String.endsWithIgnoreCase lowerName subDir.FullName) with
+        //| Some x -> x
+        //| None ->
+        //    traceWarnfn "The following directories exists:"
+        //    di.GetDirectories() |> Seq.iter (fun d -> traceWarnfn "  %s" d.FullName)
+        //
+        //    failwithf "Package directory for package %O was not found." packageName
```

Comment:
cleanup, strange why this was so much code in the past? Maybe I broke something?